### PR TITLE
[AIRFLOW-XXX] Display Azure integration in table

### PR DESCRIPTION
**Pull request by mik-laj**
This PR is one of a series about restructuring integration.rst
https://issues.apache.org/jira/browse/AIRFLOW-5431

---

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: docs/integration.rst [hunk 1]
```diff
@@ -38,80 +36,78 @@ Airflow can be configured to read and write task logs in Azure Blob Storage.
...
+   * - Service name
+     - Hook
+     - Operators
+     - Sensors
 
-* :mod:`airflow.contrib.operators.azure_cosmos_operator`
-* :mod:`airflow.contrib.sensors.azure_cosmos_sensor`
+   * - `Azure Blob Storage <https://azure.microsoft.com/en-us/services/storage/blobs/>`__
+     -
+     - :mod:`airflow.contrib.operators.wasb_delete_blob_operator`,
```
Comment (feluelle) [inline docs/integration.rst]:
```suggestion
     - :mod:`airflow.contrib.operators.wasb_delete_blob_operator`
```

**Review by feluelle — CHANGES_REQUESTED** [review]

File: docs/integration.rst [hunk 2]
```diff
@@ -38,80 +36,78 @@ Airflow can be configured to read and write task logs in Azure Blob Storage.
...
+     - :mod:`airflow.contrib.sensors.wasb_sensor`
 
-They also use :class:`airflow.contrib.hooks.azure_cosmos_hook.AzureCosmosDBHook` to communicate with Microsoft Azure.
+   * - `Azure Container Instances <https://azure.microsoft.com/en-us/services/container-instances/>`__
+     - :mod:`airflow.contrib.hooks.azure_container_instance_hook`
```
Comment (feluelle) [inline docs/integration.rst]:
```suggestion
     - :mod:`airflow.contrib.hooks.azure_container_instance_hook`,
```

File: docs/integration.rst [hunk 3]
```diff
@@ -38,80 +36,78 @@ Airflow can be configured to read and write task logs in Azure Blob Storage.
...
+       :mod:`airflow.contrib.hooks.azure_container_registry_hook`
```
Comment (feluelle) [inline docs/integration.rst]:
```suggestion
       :mod:`airflow.contrib.hooks.azure_container_registry_hook`,
```

**Review by feluelle — COMMENTED** [review]
Actually when do you use a comma and when do you just make a new line (`-`)?

File: docs/integration.rst (hunk 1 above)
Comment (mik-laj) [reply docs/integration.rst]: Thanks. Fixed.

File: docs/integration.rst (hunk 2 above)
Comment (mik-laj) [reply docs/integration.rst]: Thanks. Fixed.

File: docs/integration.rst (hunk 3 above)
Comment (mik-laj) [reply docs/integration.rst]: Thanks. Fixed.

**Review by feluelle — APPROVED** [review]

**Pull request merged**